### PR TITLE
Test operator using a file-based catalog

### LINKED ISSUE #57: Switch to using file-based catalogs for tests
The OperatorHub catalog has switched to using a file-based catalog, which breaks our existing tests that rely on `opm index add` [1][2].
The file based catalog will let us compose a smaller catalog that contains just Shipwright and Tekton.

[1] https://olm.operatorframework.io/docs/reference/file-based-catalogs/
[2] https://github.com/k8s-operatorhub/community-operators/discussions/505

/kind cleanup

### PR DESCRIPTION
**Pull request by adambkaplan**
# Changes

- Created a file-based catalog using existing manifests for Shipwright and
  Tekton releases on operatorhub.io.
- Add script to render a candidate OLM catalog using the file based
  structure that includes the built bundle image.
- Update OLM subscription to use the "candidate" channel from the new
  file-based catalog.
- Add output directory to .gitignore
- Add documentation for OLM testing
- Bump operator version to 0.8.1-snapshot
- Restore the "run operator with catalog" step in CI

Fixes #57 

/kind cleanup

# Submitter Checklist

- [x] Includes tests if functionality changed/was added
- [x] Includes docs if changes are user-facing
- [x] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)  
- [x] Release notes block has been filled in, or marked NONE

See [the contributor guide](https://github.com/shipwright-io/build/blob/master/CONTRIBUTING.md)
for details on coding conventions, github and prow interactions, and the code review process.

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: hack/render-candidate-catalog.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# 1. The candidate shipwright-operator image (from a local/CI build)
+# 2. The existing Tekton and Shipwright operators that are released on OperatorHub.io
+#
+# The core of the catalog are JSON manifests contained in `test/catalog`.
+# This script copies those manifests to _output/catalog, then does a bit of opm and sed
+# manipulation to create catalog entries for the candidate operator deployment.
+
+set -e
+set -u
+set -o pipefail
```
Comment (SaschaSchwarze0) [inline hack/render-candidate-catalog.sh:14]:
```suggestion
set -euo pipefail
```

File: hack/render-candidate-catalog.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#! /usr/bin/env bash
```
Comment (SaschaSchwarze0) [inline hack/render-candidate-catalog.sh:1]: Copyright missing.

File: hack/render-candidate-catalog.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+channelSpec="_output/catalog/candidate/shipwright-operator-channel-candidate.json"
+bundleSpec="_output/catalog/candidate/shipwright-operator-bundle.json"
+
+# Initialize the new catalog
+rm -f "_output/catalog.Dockerfile"
+rm -rf "_output/catalog"
+cp -r test/catalog _output/
+
+# Replace the placeholder bundle name for the "candidate" OLM subscription channel
+${SED_BIN} -i -E 's|"name": "shipwright-operator-latest"|"name": "shipwright-operator.v'"${VERSION}"'"|g' "${channelSpec}"
```
Comment (SaschaSchwarze0) [inline hack/render-candidate-catalog.sh:29]: I think you refer here to `CSV_VERSION` as specified in Makefile:290.

```suggestion
${SED_BIN} -i -E 's|"name": "shipwright-operator-latest"|"name": "shipwright-operator.v'"${CSV_VERSION}"'"|g' "${channelSpec}"
```

**Review by SaschaSchwarze0 — CHANGES_REQUESTED** [review]

File: hack/render-candidate-catalog.sh (hunk 2 above)
Comment (adambkaplan) [reply hack/render-candidate-catalog.sh:1]: fixed in latest push

**Review by adambkaplan — COMMENTED** [review]
@SaschaSchwarze0 Incorporated your changes, and added a few fixes to get the e2e's working with Github Actions

**Review by SaschaSchwarze0 — APPROVED** [review]
/lgtm

**Review by adambkaplan — COMMENTED** [review]
/approve

**Pull request merged**